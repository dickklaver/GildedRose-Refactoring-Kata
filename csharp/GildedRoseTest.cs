using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;

//using ApprovalUtilities;

using NUnit.Framework;

using System.Collections.Generic;
using System.Linq;

namespace csharp
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private IEnumerable<string> names;
        private IEnumerable<int> sellIns;
        private IEnumerable<int> qualities;

        public GildedRoseTest()
        {
            names = new[] { "foo", "bar", "baz" };
            sellIns = Enumerable.Range(-2, 15);
            qualities = Enumerable.Range(-2, 15);
        }
        [Test]
        public void foo()
        {
            CombinationApprovals.VerifyAllCombinations(TestFoo, names, sellIns, qualities);
        }

        [Test]
        public void brie()
        {
            CombinationApprovals.VerifyAllCombinations(TestBrie, sellIns, qualities);
        }

        [Test]
        public void backstage()
        {
            CombinationApprovals.VerifyAllCombinations(TestBackstage, sellIns, qualities);
        }

        [Test]
        public void conjured()
        {
            CombinationApprovals.VerifyAllCombinations(TestConjured, names, sellIns, qualities);
        }

        [Test]
        public void sulfuras()
        {
            CombinationApprovals.VerifyAllCombinations(TestSulfuras, sellIns);
        }

        private string TestFoo(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new GeneralItem(name, sellIn, quality) };
            return TestIt(Items);
        }

        private string TestBrie(int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new AgedBrie(sellIn, quality) };
            return TestIt(Items);
        }

        private string TestBackstage(int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new BackstagePass(sellIn, quality) };
            return TestIt(Items);
        }

        private string TestConjured(string name, int sellIn, int quality)
        {
            IList<Item> Items = new List<Item> { new Conjured(name, sellIn, quality) };
            return TestIt(Items);
        }

        private string TestSulfuras(int sellIn)
        {
            IList<Item> Items = new List<Item> { new Sulfuras(sellIn) };
            return TestIt(Items);
        }

        private static string TestIt(IList<Item> Items)
        {
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            return Items[0].ToString();
        }
    }
}
