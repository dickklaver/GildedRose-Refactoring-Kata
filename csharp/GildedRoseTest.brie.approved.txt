[-2,-2] => Aged Brie, -3, 1
[-2,-1] => Aged Brie, -3, 1
[-2,0] => Aged Brie, -3, 2
[-2,1] => Aged Brie, -3, 3
[-2,2] => Aged Brie, -3, 4
[-2,3] => Aged Brie, -3, 5
[-2,4] => Aged Brie, -3, 6
[-2,5] => Aged Brie, -3, 7
[-2,6] => Aged Brie, -3, 8
[-2,7] => Aged Brie, -3, 9
[-2,8] => Aged Brie, -3, 10
[-2,9] => Aged Brie, -3, 11
[-2,10] => Aged Brie, -3, 12
[-2,11] => Aged Brie, -3, 13
[-2,12] => Aged Brie, -3, 14
[-1,-2] => Aged Brie, -2, 1
[-1,-1] => Aged Brie, -2, 1
[-1,0] => Aged Brie, -2, 2
[-1,1] => Aged Brie, -2, 3
[-1,2] => Aged Brie, -2, 4
[-1,3] => Aged Brie, -2, 5
[-1,4] => Aged Brie, -2, 6
[-1,5] => Aged Brie, -2, 7
[-1,6] => Aged Brie, -2, 8
[-1,7] => Aged Brie, -2, 9
[-1,8] => Aged Brie, -2, 10
[-1,9] => Aged Brie, -2, 11
[-1,10] => Aged Brie, -2, 12
[-1,11] => Aged Brie, -2, 13
[-1,12] => Aged Brie, -2, 14
[0,-2] => Aged Brie, -1, 1
[0,-1] => Aged Brie, -1, 1
[0,0] => Aged Brie, -1, 2
[0,1] => Aged Brie, -1, 3
[0,2] => Aged Brie, -1, 4
[0,3] => Aged Brie, -1, 5
[0,4] => Aged Brie, -1, 6
[0,5] => Aged Brie, -1, 7
[0,6] => Aged Brie, -1, 8
[0,7] => Aged Brie, -1, 9
[0,8] => Aged Brie, -1, 10
[0,9] => Aged Brie, -1, 11
[0,10] => Aged Brie, -1, 12
[0,11] => Aged Brie, -1, 13
[0,12] => Aged Brie, -1, 14
[1,-2] => Aged Brie, 0, 0
[1,-1] => Aged Brie, 0, 0
[1,0] => Aged Brie, 0, 1
[1,1] => Aged Brie, 0, 2
[1,2] => Aged Brie, 0, 3
[1,3] => Aged Brie, 0, 4
[1,4] => Aged Brie, 0, 5
[1,5] => Aged Brie, 0, 6
[1,6] => Aged Brie, 0, 7
[1,7] => Aged Brie, 0, 8
[1,8] => Aged Brie, 0, 9
[1,9] => Aged Brie, 0, 10
[1,10] => Aged Brie, 0, 11
[1,11] => Aged Brie, 0, 12
[1,12] => Aged Brie, 0, 13
[2,-2] => Aged Brie, 1, 0
[2,-1] => Aged Brie, 1, 0
[2,0] => Aged Brie, 1, 1
[2,1] => Aged Brie, 1, 2
[2,2] => Aged Brie, 1, 3
[2,3] => Aged Brie, 1, 4
[2,4] => Aged Brie, 1, 5
[2,5] => Aged Brie, 1, 6
[2,6] => Aged Brie, 1, 7
[2,7] => Aged Brie, 1, 8
[2,8] => Aged Brie, 1, 9
[2,9] => Aged Brie, 1, 10
[2,10] => Aged Brie, 1, 11
[2,11] => Aged Brie, 1, 12
[2,12] => Aged Brie, 1, 13
[3,-2] => Aged Brie, 2, 0
[3,-1] => Aged Brie, 2, 0
[3,0] => Aged Brie, 2, 1
[3,1] => Aged Brie, 2, 2
[3,2] => Aged Brie, 2, 3
[3,3] => Aged Brie, 2, 4
[3,4] => Aged Brie, 2, 5
[3,5] => Aged Brie, 2, 6
[3,6] => Aged Brie, 2, 7
[3,7] => Aged Brie, 2, 8
[3,8] => Aged Brie, 2, 9
[3,9] => Aged Brie, 2, 10
[3,10] => Aged Brie, 2, 11
[3,11] => Aged Brie, 2, 12
[3,12] => Aged Brie, 2, 13
[4,-2] => Aged Brie, 3, 0
[4,-1] => Aged Brie, 3, 0
[4,0] => Aged Brie, 3, 1
[4,1] => Aged Brie, 3, 2
[4,2] => Aged Brie, 3, 3
[4,3] => Aged Brie, 3, 4
[4,4] => Aged Brie, 3, 5
[4,5] => Aged Brie, 3, 6
[4,6] => Aged Brie, 3, 7
[4,7] => Aged Brie, 3, 8
[4,8] => Aged Brie, 3, 9
[4,9] => Aged Brie, 3, 10
[4,10] => Aged Brie, 3, 11
[4,11] => Aged Brie, 3, 12
[4,12] => Aged Brie, 3, 13
[5,-2] => Aged Brie, 4, 0
[5,-1] => Aged Brie, 4, 0
[5,0] => Aged Brie, 4, 1
[5,1] => Aged Brie, 4, 2
[5,2] => Aged Brie, 4, 3
[5,3] => Aged Brie, 4, 4
[5,4] => Aged Brie, 4, 5
[5,5] => Aged Brie, 4, 6
[5,6] => Aged Brie, 4, 7
[5,7] => Aged Brie, 4, 8
[5,8] => Aged Brie, 4, 9
[5,9] => Aged Brie, 4, 10
[5,10] => Aged Brie, 4, 11
[5,11] => Aged Brie, 4, 12
[5,12] => Aged Brie, 4, 13
[6,-2] => Aged Brie, 5, 0
[6,-1] => Aged Brie, 5, 0
[6,0] => Aged Brie, 5, 1
[6,1] => Aged Brie, 5, 2
[6,2] => Aged Brie, 5, 3
[6,3] => Aged Brie, 5, 4
[6,4] => Aged Brie, 5, 5
[6,5] => Aged Brie, 5, 6
[6,6] => Aged Brie, 5, 7
[6,7] => Aged Brie, 5, 8
[6,8] => Aged Brie, 5, 9
[6,9] => Aged Brie, 5, 10
[6,10] => Aged Brie, 5, 11
[6,11] => Aged Brie, 5, 12
[6,12] => Aged Brie, 5, 13
[7,-2] => Aged Brie, 6, 0
[7,-1] => Aged Brie, 6, 0
[7,0] => Aged Brie, 6, 1
[7,1] => Aged Brie, 6, 2
[7,2] => Aged Brie, 6, 3
[7,3] => Aged Brie, 6, 4
[7,4] => Aged Brie, 6, 5
[7,5] => Aged Brie, 6, 6
[7,6] => Aged Brie, 6, 7
[7,7] => Aged Brie, 6, 8
[7,8] => Aged Brie, 6, 9
[7,9] => Aged Brie, 6, 10
[7,10] => Aged Brie, 6, 11
[7,11] => Aged Brie, 6, 12
[7,12] => Aged Brie, 6, 13
[8,-2] => Aged Brie, 7, 0
[8,-1] => Aged Brie, 7, 0
[8,0] => Aged Brie, 7, 1
[8,1] => Aged Brie, 7, 2
[8,2] => Aged Brie, 7, 3
[8,3] => Aged Brie, 7, 4
[8,4] => Aged Brie, 7, 5
[8,5] => Aged Brie, 7, 6
[8,6] => Aged Brie, 7, 7
[8,7] => Aged Brie, 7, 8
[8,8] => Aged Brie, 7, 9
[8,9] => Aged Brie, 7, 10
[8,10] => Aged Brie, 7, 11
[8,11] => Aged Brie, 7, 12
[8,12] => Aged Brie, 7, 13
[9,-2] => Aged Brie, 8, 0
[9,-1] => Aged Brie, 8, 0
[9,0] => Aged Brie, 8, 1
[9,1] => Aged Brie, 8, 2
[9,2] => Aged Brie, 8, 3
[9,3] => Aged Brie, 8, 4
[9,4] => Aged Brie, 8, 5
[9,5] => Aged Brie, 8, 6
[9,6] => Aged Brie, 8, 7
[9,7] => Aged Brie, 8, 8
[9,8] => Aged Brie, 8, 9
[9,9] => Aged Brie, 8, 10
[9,10] => Aged Brie, 8, 11
[9,11] => Aged Brie, 8, 12
[9,12] => Aged Brie, 8, 13
[10,-2] => Aged Brie, 9, 0
[10,-1] => Aged Brie, 9, 0
[10,0] => Aged Brie, 9, 1
[10,1] => Aged Brie, 9, 2
[10,2] => Aged Brie, 9, 3
[10,3] => Aged Brie, 9, 4
[10,4] => Aged Brie, 9, 5
[10,5] => Aged Brie, 9, 6
[10,6] => Aged Brie, 9, 7
[10,7] => Aged Brie, 9, 8
[10,8] => Aged Brie, 9, 9
[10,9] => Aged Brie, 9, 10
[10,10] => Aged Brie, 9, 11
[10,11] => Aged Brie, 9, 12
[10,12] => Aged Brie, 9, 13
[11,-2] => Aged Brie, 10, 0
[11,-1] => Aged Brie, 10, 0
[11,0] => Aged Brie, 10, 1
[11,1] => Aged Brie, 10, 2
[11,2] => Aged Brie, 10, 3
[11,3] => Aged Brie, 10, 4
[11,4] => Aged Brie, 10, 5
[11,5] => Aged Brie, 10, 6
[11,6] => Aged Brie, 10, 7
[11,7] => Aged Brie, 10, 8
[11,8] => Aged Brie, 10, 9
[11,9] => Aged Brie, 10, 10
[11,10] => Aged Brie, 10, 11
[11,11] => Aged Brie, 10, 12
[11,12] => Aged Brie, 10, 13
[12,-2] => Aged Brie, 11, 0
[12,-1] => Aged Brie, 11, 0
[12,0] => Aged Brie, 11, 1
[12,1] => Aged Brie, 11, 2
[12,2] => Aged Brie, 11, 3
[12,3] => Aged Brie, 11, 4
[12,4] => Aged Brie, 11, 5
[12,5] => Aged Brie, 11, 6
[12,6] => Aged Brie, 11, 7
[12,7] => Aged Brie, 11, 8
[12,8] => Aged Brie, 11, 9
[12,9] => Aged Brie, 11, 10
[12,10] => Aged Brie, 11, 11
[12,11] => Aged Brie, 11, 12
[12,12] => Aged Brie, 11, 13
